### PR TITLE
Remove background, box-shadow and border-bottom from author image (9593)

### LINKED ISSUE #9593: Image as link has horizontal line in background
## Description

The authors image in a blogpost is a link. It inherits link tag stylings which draws a border in the background. 

### Steps to reproduce

Open a [blogpost](https://www.gatsbyjs.org/blog/2018-10-16-why-mobile-performance-is-crucial/) and hover over the authors image.

This is how it looks.
bildschirmfoto 2018-10-31 um 08 45 28

### Expected result

No borders in background for images in link tags.

### Actual result

Border in background.

### Environment

production.

### PR DESCRIPTION
**Pull request by gurtjun**
Fixes #9593

### CODE REVIEW

File: www/src/templates/template-blog-post.js [hunk 1]
```diff
@@ -146,7 +146,16 @@ class BlogPostTemplate extends React.Component {
                   flex: `0 0 auto`,
                 }}
               >
-                <Link to={post.frontmatter.author.fields.slug}>
+                <Link
+                  to={post.frontmatter.author.fields.slug}
+                  css={{
+                    borderBottom: `0 !important`,
```
Comment (DSchau) [inline www/src/templates/template-blog-post.js]: Do all of these _need_ `!important`? I wouldn't have thought so, they shouldn't need extra specificity here.

Comment (gurtjun) [reply www/src/templates/template-blog-post.js]: Without the `!important` it's overridden by other CSS. Not sure how we can set the emotion-js inline CSS as default.

Comment (DSchau) [reply www/src/templates/template-blog-post.js]: It's here -> https://github.com/gatsbyjs/gatsby/blob/master/www/src/utils/typography.js#L169

So (annoyingly) what's happening here is that .main-body a is _more specific_ than what emotion is doing, which will be like `.asdf-1234 { borderBottom: 0; }`

Can we solve this problem without using `!important`? One way you _should_ be able to do it is hoist the css up a level, and just use something like

```js
css={{
  a: {
    borderBottom: 0,
    boxShadow: 'none',
    ':hover': {
      background: 'none'
    }
  }
}}
```

Comment (DSchau) [reply www/src/templates/template-blog-post.js]: I guess `!important` is fine too. It just sorta negates some of the benefit of using CSS in JS in the first place, hah.

Comment (gurtjun) [reply www/src/templates/template-blog-post.js]: Another solution might be using the size of the avatar image for the link. I updated the code, no `!important` needed anymore. 
If this is ok? 😄

Comment (fk) [reply www/src/templates/template-blog-post.js]: Most of the times something like https://github.com/gatsbyjs/gatsby/blob/fc81ecaa4e4b2b0b0b94dfb9337aad5866569726/www/src/components/package-readme.js#L34-L49 should work fine, e.g. 

```javascript
css={{
  "&&": {
  ...styles where specificity needs to be increased
}}
´´´

See https://github.com/emotion-js/emotion/issues/724#issue-332088759 for another example + generated `className`.

Comment (DSchau) [reply www/src/templates/template-blog-post.js]: Nice trick @fk!! @gurtjun would you want to try that?

Comment (gurtjun) [reply www/src/templates/template-blog-post.js]: Good catch @fk! That seems to work. Used the ampersands in f66f114

**Pull request merged**